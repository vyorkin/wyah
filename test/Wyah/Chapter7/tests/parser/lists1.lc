-- Lists

let nil x = x;
let cons x y = pair F (pair x y);
let null z = z T;
let head z = fst (snd z);
let tail z = snd (snd z);
let indx xs n = head (n tail xs);
