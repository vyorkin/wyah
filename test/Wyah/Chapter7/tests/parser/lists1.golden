Program [Decl "nil" (ELam (Var "x") (EVar (Var "x"))),Decl "cons" (ELam (Var "x") (ELam (Var "y") (EApp (EApp (EVar (Var "pair")) (EVar (Var "F"))) (EApp (EApp (EVar (Var "pair")) (EVar (Var "x"))) (EVar (Var "y")))))),Decl "null" (ELam (Var "z") (EApp (EVar (Var "z")) (EVar (Var "T")))),Decl "head" (ELam (Var "z") (EApp (EVar (Var "fst")) (EApp (EVar (Var "snd")) (EVar (Var "z"))))),Decl "tail" (ELam (Var "z") (EApp (EVar (Var "snd")) (EApp (EVar (Var "snd")) (EVar (Var "z"))))),Decl "indx" (ELam (Var "xs") (ELam (Var "n") (EApp (EVar (Var "head")) (EApp (EApp (EVar (Var "n")) (EVar (Var "tail"))) (EVar (Var "xs"))))))]